libpod/kube.go

```diff
@@ -253,7 +253,7 @@ func (p *Pod) podWithContainers(ctx context.Context, containers []*Container, po
 			// We add the original port declarations from the libpod infra container
 			// to the first kubernetes container description because otherwise we loose
 			// the original container/port bindings.
-			if first && len(ports) > 0 {
+			if first && len(ports) > 0 && !isInit {
```

Comment:
Do we have an issue, if there is only one init container and no primary?

Reply:
It will apply it to the first init container then